### ISSUE TITLE
Fragment cache + array payload compounds non-cached keys

### ISSUE THREAD

**Issue by apolzon** [rails-api/active_model_serializers#904]
If you define a serializer with a fragment cache, such as

```ruby
class TestSerializer < ActiveModel::Serializer
  cache key: "test", except: [:field1]
  attributes :field1, :field2
end
```

and then render in your controller, an array:
```ruby
render json: resources, each_serializer: TestSerializer
```

the fragment cache's call to attribute (line 52) will continually add field1, such that if resources has 20 objects, by the end, field1 will be called 20 times for the last invocation of the TestSerializer.

I believe ActiveModel::Serializer.attribute should only concat the key onto @_attributes if it is not already present.
i.e., line 45 (in master) should be:
```ruby
      @_attributes.concat [key] unless @_attributes.include? key
```

Also, since the @_attributes variable is at the class level, subsequent requests compound the issue even further.

**Comment by groyoh** [comment]
Nice catch! It was indeed a bug which would also happen simply by calling multiple times `.attributes` or `.attribute` with the same attribute name. I don't know if this has any consequence on the behavior itself, but might slow the overall system since we iterate over the `@_attributes` variable when calling `#attributes`.

I created a PR to fix this (#914).

**Comment by joaomdmoura** [comment]
Tks @apolzon and @groyoh !
Indeed it was a great catch, and an important issue.
But it didn't impacted the performance so much, it remained the same. :smile: Just ran some benchmark test against an older master commit  and also against ```0.10-stable``` 
:heart:
